src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -596,7 +597,9 @@ private bool TakeStartLine(SocketInput input)
                 {
                     return false;
                 }
-                var requestUri = begin.GetString(scan);
+
+                var beginPath = begin;
+                var endPath = scan;
```

Comment:
Agree on seek the `%` to help skip unnecessary decoding. Is `+` really necessary?